### PR TITLE
Manage spendable outputs generated by LDK

### LINKED ISSUE #846: Handle persisted spendable outputs
By default we handle these immediately as LDK generates `SpendableOutputs` events, but we persist them in case the corresponding transactions don't get confirmed.

I propose to:
- On node startup, delete from the database all the spendable outputs which have reached sufficient confirmations.
- On node startup, try to broadcast claim transactions for those which are not sufficiently confirmed.

We might want to handle this differently between app and coordinator, but theoretically we should be able to implement it directly in `ln-dlc-node` for both.

---

Related to #798, #844 and #845.

### PR DESCRIPTION
**Pull request by luckysori**
Fixes #846.

For all `lnd_dlc_nod::Node`s we spin up a task which ensures that LDK spendable outputs that are "owned" by LDK are moved to the `Node`s on-chain wallet.

This functionality could be split between coordinator and app (the two types of `Node` currently conceived), but it does not seem worth it at this stage.

### CODE REVIEW

File: crates/ln-dlc-node/src/node/mod.rs [hunk 1]
```diff
@@ -679,6 +684,45 @@ where
             remote_handle
         };
 
+        tokio::spawn({
```
Comment (holzeis) [inline crates/ln-dlc-node/src/node/mod.rs:687]: 🤔 I guess a task is working for the coordinator, but for the mobile app we may need to trigger it on startup? or allow for manually triggering that sync? I don't think users will have the app open for 30 minutes.

File: mobile/native/src/db/models.rs [hunk 2]
```diff
@@ -1557,6 +1573,17 @@ pub mod test {
...
         assert_eq!(expected, loaded);
+
+        // Verify that we can delete the right spendable output based on its outpoint
+        SpendableOutputQueryable::delete(outpoint, &mut connection).unwrap();
+
+        match SpendableOutputQueryable::get(outpoint, &mut connection) {
+            // Could not find spendable output as expected
+            Err(Error::NotFound) => {}
+            _ => {
+                panic!("Expected to not be able to find deleted spendable output");
```
Comment (holzeis) [inline mobile/native/src/db/models.rs:1584]: ❓ why do we need to panic here?

Comment (luckysori) [reply mobile/native/src/db/models.rs:1584]: I just followed the pattern of the other test where we delete a row. The point is that we are expecting a specific error when loading a deleted row, so we can fail the test in all other cases.

File: crates/ln-dlc-node/src/node/mod.rs (hunk 1 above)
Comment (luckysori) [reply crates/ln-dlc-node/src/node/mod.rs:687]: This is triggered on start-up inherently 😄

I agree that exposing a different mechanism for app users would also be useful. But for now we can just tell them to restart the app if they want to ensure that missing funds are claimed? I didn't want to get into adding stuff to the frontend with this change.

**Review by holzeis — APPROVED** [review]

**Review by da-kami — APPROVED** [review]

**Comment by holzeis** [comment]
// note, force pushed due to adding an entry to the changelog

**Pull request merged**